src/WebServer.Logging/ILogger.cs

```diff
@@ -0,0 +1,25 @@
+using System;
+
+namespace Restup.WebServer.Logging
+{
+    public interface ILogger
```

Comment:
I haven't come across overloads with Funcs in the most popular logging frameworks (e.g. log4net / nlog). 

Maybe it would be good to add IsTraceEnabled(), IsInfoEnabled() and so on the interface?  This is what log4net and nlog generally do and then people could use that when they're concerned about performance when generating the trace log line.